plugins/callback/foreman.py

```diff
@@ -281,17 +389,37 @@ def send_reports(self, stats):
                 report['config_report']['status']['pending'] = total['changed']
                 report['config_report']['status']['applied'] = 0
 
-            self._send_data('report', host, report)
+            self._send_data('report', 'foreman', host, report)
 
             self.items[host] = []
 
+    def send_reports(self, stats):
+        if self.report_type == "foreman":
+            self.send_reports_foreman_legacy(stats)
+        elif self.report_type == "proxy":
+            self.send_reports_proxy_host_report(stats)
+        else:
+            self._display.warning(u'Unknown foreman endpoint type: {type}'.format(type=self.report_type))
+
+    def drop_nones(self, d: dict) -> dict:
+        """Recursively drop Nones in dict d and return a new dict"""
+        dd = {}
+        for k, v in d.items():
+            if isinstance(v, dict):
+                dd[k] = self.drop_nones(v)
+            elif isinstance(v, (list, set, tuple)):
+                dd[k] = type(v)(self.drop_nones(vv) if isinstance(vv, dict) else vv
+                                for vv in v)
+            elif v is not None:
+                dd[k] = v
+        return dd
+
     def append_result(self, result, failed=False):
-        name = result._task.get_name()
```

Comment:
I guess for better UI we could keep the name of a task, but it's up to you how to send it exactly: put this into `result._result` or send it in the old way.

Reply:
Great find! Yeah I was confused I thought it is the same as module name but it can be different.

I stored it into the hash under "name" key, I hope there is not such key in use (in my few examples I haven't seen such key used).